File: 1-easy/problems/1160-find-words-that-can-be-formed-by-characters/solution.ts
function countCharacters(words: any, chars: any): boolean | number | string | any {
  """
  :type words: List[str]
  :type chars: str
  :rtype: int
  """
  def check(word, chars, count):
      if ((word).length > (chars).length) {
          return false;
      curr_count = collections.Counter()
      for (c in word) {
          curr_count[c] += 1
          if (c not in count || count[c] < curr_count[c]) {
              return false;
      return true;

  count = collections.Counter(chars)
  return sum((word).length for word in words if check(word, chars, count));
}

export { countCharacters };

File: 1-easy/problems/1160-find-words-that-can-be-formed-by-characters/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def countCharacters(self, words, chars):
        """
        :type words: List[str]
        :type chars: str
        :rtype: int
        """
        def check(word, chars, count):
            if len(word) > len(chars):
                return False
            curr_count = collections.Counter()
            for c in word:
                curr_count[c] += 1
                if c not in count or count[c] < curr_count[c]:
                    return False
            return True
        
        count = collections.Counter(chars)
        return sum(len(word) for word in words if check(word, chars, count))
